### ISSUE TITLE
Is Swift 5 support planned ?

### ISSUE THREAD

**Issue by AlaeddineOuerteni** [Moya/Moya#1863]

**Comment by sunshinejr** [comment]
Hey @AlaeddineOuerteni. Version 13 of Moya is actually using Swift 5! The problem is that all of the dependencies didn't transition to Swift 5 when we were releasing Moya 13 and so we only did an update to our codebase. Version 14 will have all of the dependencies moved to Swift 5 as well.

**Comment by BigPoro** [comment]
> Hey @AlaeddineOuerteni. Version 13 of Moya is actually using Swift 5! The problem is that all of the dependencies didn't transition to Swift 5 when we were releasing Moya 13 and so we only did an update to our codebase. Version 14 will have all of the dependencies moved to Swift 5 as well.

Is Moya/RxSwify sypport Swift 5?

**Comment by 913868456** [comment]
RxSwift have version 5.0,but Moya still user 4.5, always have this problem
```
Downloading RxSwift.framework binary at "[4,5]"
***  Skipped installing RxSwift.framework binary due to the error:
	"Incompatible Swift version - framework was built with 5.0 (swiftlang-1001.0.69.5 clang-1001.0.46.3) and the local version is 5.0.1 (swiftlang-1001.0.82.4 clang-1001.0.46.5)."
```
I want to know how to solve this problem?

**Comment by sunshinejr** [comment]
@BigPoro Moya/RxSwift 13.0.* has a core built with Swift 5, but RxSwift version 4.5. However, there is an alpha version for Moya, 14.0.0-alpha.1, that has both ReactiveSwift 6 and RxSwift 5 in it if you want to try full Swift 5.0 support.

@913868456 seems like you are building the carthage frameworks with binaries enabled. These binaries are uploaded by maintainers to have a quick access to the framework compiled with given Swift version. Seems like RxSwift 4.5 was uploaded and compiled by Swift 5.0, and you have 5.0.1. When this happens, just use flag `--no-use-binaries` for Carthage and it will ignore the uploaded binary and recompile it using your own Swift version and tooling:
```bash
carthage bootstrap --platform iOS --no-use-binaries
```

Let me know if I can help you further :)

**Comment by 913868456** [comment]
Thanks, I flow the steps you recommended, solve the problem.

**Comment by witchnana** [comment]
Will moya 14.0.0 support alamofire 5.0.0?

**Comment by sunshinejr** [comment]
Hey @witchnana! Yes, the release is centered around bumping all our dependencies, including Alamofire 5. They are still in beta so we are waiting for final API changes (as they mentioned that these could happen in beta as they as they are not sticking to semver during 5.0 beta).

**Comment by larryonoff** [comment]
@sunshinejr I think it would be great release some intermediate version (e.g. 13.1) of Moya with the latest (Swift 5.1) RxSwift / ReactiveSwift version. It's not a big deal to do. I can do PR.

PS. I already looking forward to migrate into ReactiveSwift 6.0 and remove Result dependency, but cannot since Moya.

**Comment by sunshinejr** [comment]
@larryonoff you mean a version where we only bump RxSwift and ReactiveSwift? These are still major updates and so we would need to make a major update as well. Anyhow I'd love to get some feedback on that from @Moya/contributors as Alamofire is taking a long time to release.

**Comment by larryonoff** [comment]
It can be marked as 14.0 since it’s major update, but bump is needed. Since it takes too long to wait new version of Moya with all of these.

**Comment by acecilia** [comment]
I am working on a new project using Swift 5. In order to avoid the Result dependency and be able to use RxSwift 5, at the moment I am using Moya straight from the latest commit on the development branch.

I understand that you would like to wait for Alamofire 5 to be out. I think that makes sense depending on the estimations that Alamofire maintainers have rewarding the ETA for the next release.

**Comment by larryonoff** [comment]
I solved the issue easily, set dependency to `Moya/Core` and put `Moya/ReactiveSwift` code into the project, but with ReactiveSwift 6.0

**Comment by sunshinejr** [comment]
It seems like the rc build of Alamofire 5 should be just around the corner: https://github.com/Alamofire/Alamofire/milestone/57. I'm gonna update the PR for Alamofire 5 and once they release rc.1 we can release our beta.1 (finally).

**Comment by MortyMerr** [comment]
Can't wait for RxSwift update to ~> 5

**Comment by larryonoff** [comment]
up

**Comment by sunshinejr** [comment]
If the rc won't happen this weekend, I think we can release new pre-release Moya anyways. For this, I'd need someone to try and review #1810, gonna mention people so we can get this done early next week.

**Comment by sunshinejr** [comment]
Alright friends, alpha.2 is released! Removed `Result`, upgraded `Alamofire` and few more cool adjustments.

![Zrzut ekranu 2019-08-1 o 17 28 45](https://user-images.githubusercontent.com/5232779/62306265-d347f500-b481-11e9-9da8-89f1b9f1e4ff.png)

See full changelog here: https://github.com/Moya/Moya/releases/tag/14.0.0-alpha.2

**Comment by MortyMerr** [comment]
Up

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by teameh** [comment]
Hi @sunshinejr is there a list of things / issues / PR's that need to be fixed before the beta / stable release of 14.0.0 ? We're eager to upgrade to 14, Moya is our last dependency that blocks using the latest RxSwift 5 😄 . 

Happy to help!

**Comment by MortyMerr** [comment]
+
I miss that `compactMap` 😭

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by sunshinejr** [comment]
@teameh not really, basically we are aligning our release with the release of Alamofire 5. So once they are done making final release, we will do that as well!

**Comment by teameh** [comment]
Check! Thanks, I'll ask around there 😄

**Comment by sunshinejr** [comment]
Alamofire 5.0.0-rc just came out so we will prepare for beta!

**Comment by teameh** [comment]
Great. We'll hop on the rc and will provide some feedback to the docs if needed when it's out :)

**Comment by sunshinejr** [comment]
Hey y'all, I just released Moya 14.0.0-beta.1 🚀

**Comment by LucianoPAlmeida** [comment]
Moya 14.0.0 is out 🚀  So we can close this one :)
